data/txntest/txn.go

```diff
@@ -117,6 +118,10 @@ func (tx *Txn) internalCopy() {
 	for i := 0; i < len(tx.Boxes); i++ {
 		tx.Boxes[i].Name = append([]byte(nil), tx.Boxes[i].Name...)
 	}
+	tx.Access = append([]transactions.ResourceRef(nil), tx.Access...)
+	for i := 0; i < len(tx.Access); i++ {
+		tx.Access[i].Box.Name = append([]byte(nil), tx.Access[i].Box.Name...)
```

Comment:
check for non-empty `tx.Access[i].Box` ?

Reply:
This is just copying the name for immutability. If it's nil, it stays nil. That's why I used the weird construct of `append` onto `nil`.  That function does it a lot, for the `Note`, `ApplicationArgs` etc.